plugin_interpreter/README.md

```diff
@@ -1 +1,85 @@
-***THIS IS A TEST***
+In order to create a plugin you must create a python script with the name of
+the the plugin. Inside this script you must create a class with the name of
+the plugin file's name inside of the plugins folder.
+
+ExamplePlugin.py
+```python
+from ..src import controller_plugin
+
+class ExamplePlugin(controller_plugin.ControllerPlugin):
+```
+
+You will need to specify the capabilities of your plugin to the database.
+you can do this either by creating a list of dictionaries containing each
+command inside the plugin file and passing it to the super's constructor,
+```python
+commands = [
+  {
+    "CommandName": "my command",
+    "Tooltip": "A helpful tooltip.",
+    "Output": True,
+    "Inputs": [
+      {
+        "Name": "task",
+        "Type": "textbox",
+        "Tooltip": "type the option you want"
+        "Value": ""
+      }
+    ],
+    "OptionalInputs":[]
+  }
+]
+def __init__(self):
+  super().__init__("ExamplePlugin", commands)
+```
+or you can place the commands in JSON format in a .json file with the same
+name as the plugin in the plugins folder and not pass a command list to the
+super's constructor.
+```python
+def __init__(self):
+  super().__init__("ExamplePlugin")
+```
+
+you **must** also override 2 functions, start() and _stop()
+```python
+def start(self, logger, signal):
+
+def _stop(self):
+```
+start() is the entrypoint of the plugin. It should have some sort of
+control loop or lead to one. You will be given a logger to log any issues
+and a ctype boolean `signal` that can be used to know when the plugin is
+being shut down by the controller. `signal` will be true if the plugin is to
+be shut down. _stop() is used for any cleanup or teardown you may need and is
+called when the plugin ends.
+
+A typical plugin will interact with 2 main Controller_Plugin methods:
+```python
+self.request_job()
+self.respond_output()
+```
+`request_job()` will return the next job and automatically update its state
+in the databse.
+`respond_output(job, output)` takes a job and the output associated with it and
+update the database with it as well as update the job's state.
+
+If you need to signify that an error has occured in plugin operation you can use
+`respond_error(job, message)` where you specify the job that errored and an
+optional error message. This will update the job's status to error.
+
+`get_file(filename)` can be used if you want to get a file uploaded by a user
+from the database.
+
```

Comment:
I would specify here that these are automatically managed, and that manually editing them would be an edge case scenario.